enhancements/sdk-external-and-pluggable-validations.md

```diff
@@ -0,0 +1,494 @@
+
+##### Removing a deprecated feature
+
+- Announce deprecation and support policy of the existing feature
+- Deprecate the feature
+
+### Upgrade / Downgrade Strategy
+
+If applicable, how will the component be upgraded and downgraded? Make sure this
+is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade in order to keep previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade in order to make use of the enhancement?
+
+### Version Skew Strategy
+
+* The version of the validators can change however the validator author sees fit.
+* The API or contract between `operator-sdk` and validators will be
+  * input to validator: bundle directory
+  * output from validator: `ManifestErrors` JSON
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation
+History`.
+
+## Drawbacks
+
+Validators would have to be their own executables which could result in a
+compilation step being needed depending on the language used to implement them.
+
+## Alternatives
+
+* put validations in their own images
+  * need to define "API" contract what is the entrypoint and what parameters do
+    we give it
+  * pro:
+    * already have a precendence for running things from images
+    * familiar tech
+  * con:
+    * would need a cluster to run these validations
+    * authors would have to create binaries of their validations anyway
+
+TODO: ---------- vvvvvv CUT vvvvvv ----------
+* wrap validations in their own executables
+  * validations would be wrapped in an executable that would be called from
+    operator-sdk
+  * pro:
+    * could reuse the existing go validations and put a main.go in front to make
+      them executable
+    * follows a similar phase 2 plugin path
+    * could reuse some of the phase 2 tech to run this
+    * would not need a cluster necessarily to run the validation
+  * con:
+    * authors would have to create binaries of their validations
+TODO: ---------- ^^^^^^ CUT ^^^^^^ ----------
+
+* use a language like JavaScript or CUE to define all validations
+  * validations could be run from a git repo, i.e. operator-sdk could pull it
+    and then evaluate it
+  * pro:
+    * simpler delivery, expose via a gitrepo and done
+  * con:
+    * all existing validations would have to be re-written in a new language
+      structure which could introduce new bugs
+    * unproven technology
+    * would have to write the engine to know how to execute these
+
+* use scorecard to do the validations
+  * create validations written in scorecard as custom tests
+  * pro:
+    * infrastructure required to run is already built within scorecard
+  * con:
+    * would need a cluster to run these validations
+
+## Infrastructure Needed [optional]
```

Comment:
WDYT about we remove the sections that we do not need to fill from the EP to try to make it less verbose?